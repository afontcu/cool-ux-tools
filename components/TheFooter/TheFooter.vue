<template>
  <footer class="Footer">
    <div class="Footer__wrapper o-wrapper u-flex u-padding-vertical-large u-text-h4 u-text-thin">
      <div class="Footer__line u-margin-bottom-small">
        Made with ðŸ’š by&nbsp;<fancy-link target="_blank" href="https://twitter.com/afontcu_" title="My Twitter" rel="noopener" >@afontcu_</fancy-link>
      </div>
      <div class="Footer__line u-margin-bottom-small">
        Feel free to report issues or ðŸ†’ ideas on <fancy-link target="_blank" href="https://github.com/afontcu/cool-ux-tools" title="Github repository" rel="noopener">Github &rarr;</fancy-link>
      </div>
    </div>
  </footer>
</template>

<script>
import FancyLink from '@/components/FancyLink'

export default {
  name: 'TheFooter',
  components: {
    FancyLink,
  },
  data () {
    return {}
  },
}
</script>

<style lang="scss">
.Footer {
  position: relative;
  color: #fff;
}

.Footer,
.Footer__wrapper {
  background-color: #2f2f2f;
}

.Footer__wrapper {
  flex-direction: column;
  justify-content: center;
}

@media screen and (min-width: $small-breakpoint) {
  .Footer__wrapper {
    flex-direction: row;
  }

  .Footer__line + .Footer__line::before {
    content: " Â· ";
    padding: 0 $global-spacing-unit;
  }
}
</style>
