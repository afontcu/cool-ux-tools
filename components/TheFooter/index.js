import TheFooter from './TheFooter'

export default TheFooter
