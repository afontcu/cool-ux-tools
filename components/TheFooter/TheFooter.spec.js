import TheFooter from './index.js'
import { mount } from '@vue/test-utils'

describe('TheFooter.vue', () => {
  it('is a component', () => {
    const wrapper = mount(TheFooter)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders main node', () => {
    const wrapper = mount(TheFooter)
    expect(wrapper.classes()).toContain('Footer')
  })

  it('renders a snapshot', () => {
    const wrapper = mount(TheFooter)
    expect(wrapper.html()).toMatchSnapshot('default')
  })
})
