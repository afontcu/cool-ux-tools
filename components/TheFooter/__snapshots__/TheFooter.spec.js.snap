// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheFooter.vue renders a snapshot: default 1`] = `

<footer class="Footer">
  <div class="Footer__wrapper o-wrapper u-flex u-padding-vertical-large u-text-h4 u-text-thin">
    <div class="Footer__line u-margin-bottom-small">
      Made with ðŸ’š by&nbsp;
      <a href="https://twitter.com/afontcu_"
         title="My Twitter"
         class="FancyLink"
         target="_blank"
         rel="noopener"
      >
        @afontcu_
      </a>
    </div>
    <div class="Footer__line u-margin-bottom-small">
      Feel free to report issues or ðŸ†’ ideas on
      <a href="https://github.com/afontcu/cool-ux-tools"
         title="Github repository"
         class="FancyLink"
         target="_blank"
         rel="noopener"
      >
        Github â†’
      </a>
    </div>
  </div>
</footer>

`;
